app/controllers/collections_controller.rb

```diff
@@ -3,6 +3,7 @@ class CollectionsController < ManageCollectionsController
 
   before_action :set_collection
   before_action :ensure_correct_collection_provider, only: [:edit, :clone, :revert, :destroy]
+  before_action :set_collection_download, only: [:download_xml]
```

Comment:
This piece of code will only be used in this one action. I don't think it needs to be a separate method, I would think we could just move the work that is done in this method into the `download_xml` method.

Reply:
I will remove the before action. I think the method should be best kept separate (based on keeping methods smaller if possible), but I don't have to.